### PR TITLE
Add documentation site base

### PR DESCRIPTION
**Pull request by jamilbk**
Adds the new docs layout created by @gongjason

### CODE REVIEW

**Comment by jamilbk** [comment]
@gongjason Does this branch still work for you? I got jekyll errors related to `{% seo %}` not being a recognized tag, then after fixing that got errors about missing `title.html` and `nav.html` files from `_includes`.

I'm sure I'm missing something

(Also removed Gemfile.lock from .gitignore -- this is a file to keep in the repo)

**Comment by gongjason** [comment]
> @gongjason Does this branch still work for you? I got jekyll errors related to `{% seo %}` not being a recognized tag, then after fixing that got errors about missing `title.html` and `nav.html` files from `_includes`.
> 
> I'm sure I'm missing something
> 
> (Also removed Gemfile.lock from .gitignore -- this is a file to keep in the repo)

@jamilbk , if you're building and testing locally you'll have to uncomment L36 and comment L35 so it's pointing at the correct theme file. There's probably a better way to do this, but I haven't looked into how to reference the right theme files based on the environment.

https://github.com/firezone/firezone/blob/081bc3c4a85fcb212b5eebc440a5e3fa73882be9/docs/_config.yml#L35-L36

**Comment by jamilbk** [comment]
> @jamilbk , if you're building and testing locally you'll have to uncomment L36 and comment L35 so it's pointing at the correct theme file. There's probably a better way to do this, but I haven't looked into how to reference the right theme files based on the environment.

Ahh, thanks for pointing this out!

**Comment by gongjason** [comment]
@jamilbk actually we can keep both lines uncommented to work for github pages and locally 😅. Made the change in https://github.com/firezone/firezone/pull/371/commits/0a98dd60ed78099993c5ce8bbccf9caab529bdee

**Comment by jamilbk** [comment]
> @jamilbk actually we can keep both lines uncommented to work for github pages and locally 😅. Made the change in [0a98dd6](https://github.com/firezone/firezone/commit/0a98dd60ed78099993c5ce8bbccf9caab529bdee)

Cool thanks for setting this up. I'm going to do a little reading about jekyll remote themes and learn a little bit more about the steps they take to ensure theme integrity -- if lacking I'll just pull the theme into this repo and commit it (this is what we're doing with the product side dashboard theme).

**Comment by jamilbk** [comment]
@gongjason I fixed a bunch of little things. Think this is ready to go. Still need to do firezone/backlog#187 after this is merged.

File: docs/_config.yml [hunk 1]
```diff
@@ -1,54 +1,82 @@
...
+url: "" # the base hostname & protocol for your site, e.g. http://example.com
+permalink: pretty
+twitter_username: firezonevpn
+github_username:  firezone
+
+# Set a path/url to a logo that will be displayed instead of the title
+logo: "/assets/images/firezone-logo-wordmark-dark.svg"
+
+# Build settings
+remote_theme: firezone/just-the-docs
```
Comment (jamilbk) [inline docs/_config.yml:35]: Changed this to link to our own repo

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+---
+
+# Overview
+
+---
+
+Firezone is an open source, self-managed VPN server and Linux firewall designed for simplicity and security.
+
+Here are a few helpful links to get you started:
+* [Quick Start Guide]({{ site.baseurl }}{% link docs/get-started/index.md %})
```
Comment (jamilbk) [inline docs/index.md:18]: Results in prettier URLs

File: docs/docs/get-started/client-setup.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+---
+
+## Adding Devices
+Add a device through the Web UI by clicking the "Add Device" button under `/devices` or `/users`. Once the device profile is created, you can send the WireGuard configuration file to the client.
+
+## Client Instructions
+Clients should download the latest version of the WireGuard client from https://www.wireguard.com/install/. Once installed, clients can import the tunnel via the `.conf` file sent by the administrator.
+
+![Client Instructions](https://user-images.githubusercontent.com/52545545/147296703-a7fa8fce-33bb-42ef-83b3-1b8e9047d1d5.gif)
```
Comment (jamilbk) [inline docs/docs/get-started/client-setup.md:18]: Fixed image link

File: docs/_sass/color_schemes/fz.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+$link-color: $fz-purple-000;
+$btn-primary-color: $fz-purple-100;
+
+// Fonts
+
+$body-font-family: "Fira Sans", sans-serif;
+$mono-font-family: "Fira Mono", monospace;
+
+h1, h2, h3, h4, h5, h6 {
```
Comment (jamilbk) [inline docs/_sass/color_schemes/fz.scss:28]: Made the titles in OpenSans

**Comment by jamilbk** [comment]
@gongjason Link to a diff of my recent changes here: https://github.com/firezone/firezone/compare/250f4e8924d742cb0e10247dfd2459af9931252a...add-documentation-site-base

File: docs/docs/get-started/client-setup.md (hunk 3 above)
Comment (gongjason) [reply docs/docs/get-started/client-setup.md:18]: Is this mainly to keep the size of the repo small? The process to update many images seems slightly more cumbersome than hosting in the repo. We would need to upload each image into GitHub user content and then find the place in the docs where we need to overwrite for each image. If we hosted in the repo we could have a static reference that I can overwrite.

**Review by gongjason — APPROVED** [review]
looks good!

File: docs/docs/get-started/client-setup.md (hunk 3 above)
Comment (jamilbk) [reply docs/docs/get-started/client-setup.md:18]: For keeping references to image URLs, maybe [jekyll variables](https://www.ayushsharma.in/2021/08/using-variables-in-jekyll-to-define-custom-content) would work well?

Yeah we want to keep the size of the repo small, and we want to keep large (>50kb) binary files out of the repository as much as possible, especially if they're expected to change often.

**Pull request merged**